crates/grafbase-sdk/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+#![doc = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/README.md"))]
+#![deny(missing_docs)]
+
+#[doc(hidden)]
+pub mod extension;
+pub mod host_io;
+pub mod types;
+
+pub use extension::{Extension, Resolver};
+pub use wit::{Error, ExtensionType, SharedContext};
+
+use semver::Version;
+use std::sync::LazyLock;
+
+/// The minimum version of Grafbase Gateway that can run this SDK.
+pub const MINIMUM_HOST_VERSION: Version = Version::new(0, 48, 0);
+
+/// The version of the SDK.
+pub static GUEST_VERSION: LazyLock<Version> =
+    LazyLock::new(|| Version::parse(std::env!("CARGO_PKG_VERSION")).expect("must be valid semver"));
+
+#[doc(hidden)]
+pub fn check_host_version(host_version: u32) -> i32 {
+    let version = unpack_version(host_version);
+
+    if version < MINIMUM_HOST_VERSION {
+        -(pack_version(&MINIMUM_HOST_VERSION) as i32)
+    } else {
+        pack_version(&GUEST_VERSION) as i32
+    }
+}
+
+fn pack_version(version: &Version) -> u32 {
+    let major = version.major as u32;
+    let minor = version.minor as u32;
+    let patch = version.patch as u32;
+
+    (major << 16) | (minor << 8) | patch
+}
+
+fn unpack_version(version: u32) -> Version {
+    let major = (version >> 16) as u64;
+    let minor = (version >> 8) as u64;
+    let patch = (version & 0xFF) as u64;
+
+    Version::new(major, minor, patch)
+}
+
+struct Component;
+
+/// Register the extension to the Grafbase Gateway. This macro must be called in the extension
+/// crate root for the local extension implementation.
+///
+/// The first parameter is the type of the extension, and the second parameter is the extension
```

Comment:
This line is confusing. Maybe link to the `ExtensionType` enum for the first, and change "the extension type" to "the struct that implements the `Extension` trait"?